src/lib/core/say.c

```diff
@@ -1033,6 +1048,60 @@ format_syslog_header(char *buf, int len, int level,
 enum { SAY_BUF_LEN_MAX = 16 * 1024 };
 static __thread char say_buf[SAY_BUF_LEN_MAX];
 
+/** To disable ANSI color support set the `NO_COLOR`*/
+static int no_color = 0;
+
+/**
+ * Wraps the content of the say_buf in ANSI color codes based on the
+ * provided level.
+ * This function is intended only for output to a terminal (tty).
+ */
+static int
+wrap_buffer(int level, int size)
+{
+	if (no_color == 1)
+		return size;
+
+	char var_buf[5];
+	const char *envvar = getenv_safe("NO_COLOR", var_buf,
+					 sizeof(var_buf));
+	if (envvar != NULL && envvar[0] != '\0') {
+		no_color = 1;
+		return size;
+	}
+
+	if (level > S_WARN || size <= 0)
+		return size;
+
+	const char *color_prefix;
+	int color_prefix_len;
+
+	if (level < S_WARN) {
+		color_prefix = ANSI_COLOR_RED;
+		color_prefix_len = sizeof(ANSI_COLOR_RED) - 1;
+	} else if (level == S_WARN) {
+		color_prefix = ANSI_COLOR_YELLOW;
+		color_prefix_len = sizeof(ANSI_COLOR_YELLOW) - 1;
+	}
```

Comment:
Some comments regarding the logging levels would be appreciated.

Reply:
added